internal/rbd/controllerserver.go

```diff
@@ -165,6 +165,7 @@ func (cs *ControllerServer) parseVolCreateRequest(
 func buildCreateVolumeResponse(req *csi.CreateVolumeRequest, rbdVol *rbdVolume) *csi.CreateVolumeResponse {
 	volumeContext := req.GetParameters()
 	volumeContext["pool"] = rbdVol.Pool
+	volumeContext["dataPool"] = rbdVol.DataPool
```

Comment:
set this only if the DataPool is not empty?

Reply:
rbdVol is set here
https://github.com/ceph/ceph-csi/blob/28fef9b3797e41f277d11d9a3bde11e971994503/internal/rbd/rbd_util.go#L1150-L1159
I think the value is safe to copy here.

Reply:
my concern is when we create PVC from storageclass without `dataPool` we should not set the `dataPool=""` in the PV object.

Reply:
Indeed, you're right.
I've added a check.